### ISSUE TITLE
[vault-secrets-webhook] pod mutation with multiple imagePullSecrets

### ISSUE THREAD

**Issue by dkulchinsky** [banzaicloud/bank-vaults#1232]
**Describe the bug**:
We have a pod with two containers, each uses an image from a private registry that has two "organizations" (e.g. `our-registry.com/org1/container-image1` & `our-registry.com/org2/container-image2`) each of these registry organizations have unique credentials, the Pod in question lists two secrets under `imagePullSecrets` (one per each "org").

K8s is able to pull images for both containers using the provided pull secrets.

However, `vault-secrets-webhook` picks the first listed `imagePullSecret`, for `org1`, where the image being pulled is from `org2`.

**Expected behaviour**:
`vault-secrets-webhook` should attempt all listed `imagePullSecrets` to find the one that works.

**Steps to reproduce the bug**:
1. create a pod with 2 containers
2. each container image should be stored in a separate private registry (container1 in org1, container2 in org2)
3. each private registry should have unique credentials
4. pod should have both credentials listed under `imagePullSecrets`, in the following order: `org1`, `org2`
5. mutation should be defined for container2 (with image in org2)
6. `vault-secrets-webhook` will attempt to fetch the image from org1 using credentials for org1 and fail

**Additional context**:
N/A

**Environment details**:
- Kubernetes version: v1.17.13-gke.2600
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): GKE
- bank-vaults version (e.g. 0.4.17): 1.9.0
- Install method (e.g. helm or static manifests): helm
- Logs from the misbehaving component (and any other relevant logs):
```
[vault-secrets-webhook-764c6dcb49-7gmpf] time="2021-01-13T21:02:45Z" level=info msg="found credentials for registry container-registry.our-domain.com in pod imagePullSecret: platform-docs/registry-org1"
[vault-secrets-webhook-764c6dcb49-7gmpf] time="2021-01-13T21:02:45Z" level=info msg="I'm using registry https://container-registry.our-domain.com"
[vault-secrets-webhook-764c6dcb49-7gmpf] time="2021-01-13T21:02:45Z" level=error msg="admission webhook error: cannot fetch image descriptor: GET https://container-registry.our-domain.com/v2/external/oauth2-proxy/manifests/v6.1.1: UNAUTHORIZED: unauthorized to access repository: external/oauth2-proxy, action: pull: unauthorized to access repository: external/oauth2-proxy, action: pull" app=vault-secrets-webhook
```

for above image, the correct pull secret is the 2nd one listed under pod's `imagePullSecrets`: `registry-org2`, however `vault-secrets-webhook` uses the first one listed `registry-org1` and fails.

- Resource definition (possibly in YAML format) that caused the issue, without sensitive data:

I suppose the only relevant bit here is the `imagePullSecrets` definition:
```
    imagePullSecrets:
    - name: registry-org1
    - name: registry-org2
```

/kind bug

**Comment by bonifaido** [comment]
Accepted, this is going to be fixed with the already planned: https://github.com/google/go-containerregistry/blob/master/pkg/authn/k8schain/k8schain.go

**Comment by dkulchinsky** [comment]
Thanks @bonifaido !

**Comment by dkulchinsky** [comment]
Hi @bonifaido, I'm cross-posting this from #1269 which seem to be a similar issue, however... it seems that the fix in #1236 didn't quite solve the problem and we're still reproducing the issue described here :-(

as a reminder, we have multiple `imagePullSecrets` defined that indeed exist and are correct (the reason is because same Pod has containers coming from different internal repos that have different secrets).

Now in this scenario, VSW is only able to pull images for the registry whose secret is the first in the `imagePullSecrets` list, if the image is in the registry whose secret is 2nd - it fails to pull the image and mutation is not performed.
